src/handlers/ResponseHandlers.ts

```diff
@@ -13,12 +13,19 @@ const noContent: ResponseHandler = {
 
 const success: ResponseHandler = {
     canProcess: isAnyOkStatus,
-    process: async <T>(response: Response): Promise<T> => await response.json(),
+    process: async <T>(response: Response): Promise<T> => {
+        if (response.headers.get('Content-Type')?.includes('application/json')) {
```

Comment:
Are we sure that all of the other APIs have the Content-Type header with `application/json`? I know they *should*, but I'm worried of this being a breaking change.

If we're not sure, we may unfortunately be better off adding an option like `responseFormat?: 'auto' | 'json' | 'text'`. It could default to 'auto', determining the type from the header. We could make it so that all the current endpoints explicitly choose `responseFormat: 'json'`.

Reply:
Alright thanks for your input, I will work on adding some response format somewhere. I think it will be better in the long run because it will allows us to ditch (deprecate) the `successBlob` handler at the same time.

Reply:
Done